### DISCUSSION TITLE
Feature Request: Content Staging / Live Preview

### DISCUSSION THREAD

**Discussion by RitterKnightCreative** [craftcms/cms#8476] [Ideas]
Saw [this](https://magento.com/campaign/magento-feature-videos-staging) on the Magento site recently and was quite impressed.

While I thought about posting this to the Commerce feedback, it's quite relevant to Craft itself. Something I've been wanting to do for awhile: being able to stage content, preview all of those draft entries,  and then push them all out at a certain point in time—automatically.  Craft already has parts of the sharing feature built-in, it would of course be a matter of extending it.

What's especially compelling about this feature is you can preview the entire site in that state. (If Commerce supported this, for example, you could test the site, promotional codes and all, before the campaign went live.)

The use cases are endless here: product launches, upcoming major news stories (eg political or sporting news where there's a winner/loser, etc.)

**Comment by sjcallender** [comment]
"Scheduled content" FTW. Especially, as Dave noted, for retailers who have seasonal content to promote. They could plan and stage the content within Craft and set a future date for that version to go live.

**Comment by davidhellmann** [comment]
We had this topic this week and this would be REALLY nice!

**Comment by lenvanessen** [comment]
Bump! Would be awesome!

**Comment by jripmeester** [comment]
Bump again. Several use cases in different projects for this.

**Comment by brandonkelly** [comment]
We do intend to address this, which is why it’s on the 5.0 milestone (note that 5.0 is not all that far away as we will be moving to a more fast-paced release cycle after 4.0 ships). it’s a major feature that will require some thinking. “Bumping” it won’t help us get to it any faster.

**Comment by rellafella** [comment]
I am really keen to see native support for this, but if anyone following this discussion is after an interim solution. The [Publisher plugin](https://github.com/Goldinteractive/craft3-publisher) has been doing the job for my team for some time now.

**Comment by mtwalsh** [comment]
Hi, we've got a client that is considering Craft CMS for one of their most highly trafficked sites, but are a little hesitant due to this feature not being available yet. Are you able to provide any update since the last news at the end of 2020?

I've put an [issue on the Publisher plugin](https://github.com/Goldinteractive/craft-publisher/issues/3) to see if that is going to be upgraded to Craft 5, as that may suffice in the meantime. _(Update: It will be updated for Craft 5)_

**Reply by thupsi** [reply]
There's a 'Content Releases' item slated for 5.x in the [public roadmap](https://craftcms.com/roadmap). No specific ETA though.

**Reply by brandonkelly** [reply]
I can’t say for sure but it’s something we are definitely planning for 5.x.

**Reply by mtwalsh** [reply]
Thanks for the responses, very much appreciated.
